### PR TITLE
Make use_pkgdown configurable

### LINKED ISSUE #392: Make use_pkgdown configurable?
Would you take a PR that implements configurable arguments to `use_pkgdown()`? I'd like to use usethis in pkgdown but these values change across different pkgdown usage scenarios.

Something like this:

```r
#' Use pkgdown
#'
#' [pkgdown](https://github.com/hadley/pkgdown) makes it easy to turn your
#' package into a beautiful website. This helper creates `_pkgdown.yml`
#' and `docs/` for you, and adds to `.Rbuildignore`
#'
#' @export
use_pkgdown <- function(pkg_file = "_pkgdown.yml", pkg_dir = "docs") {
  check_is_package("use_pkgdown()")
  edit_file(proj_path(pkg_file))
  use_build_ignore(pkg_file)

  use_directory(pkg_dir, ignore = TRUE)
}
```

### PR DESCRIPTION
**Pull request by jayhesselberth**
Closes #392

### CODE REVIEW

**Comment by jayhesselberth** [comment]
`build_dir` could also be `dst_path`, which is what pkgdown uses.

File: R/pkgdown.R [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+#' and a directory for pkgdown docs, and adds them to `.Rbuildignore`
+#'
+#' @seealso http://pkgdown.r-lib.org/articles/pkgdown.html#configuration
+#'
+#' @param config_file pkgdown yaml config file
+#' @param dst_path target directory for pkgdown docs
 #'
 #' @export
-use_pkgdown <- function() {
+use_pkgdown <- function(config_file = "_pkgdown.yml", dst_path = "docs") {
```
Comment (hadley) [inline R/pkgdown.R]: If the user sets `dst_path` should it be reflected in the generated `pkgdown.yml` file?

File: R/pkgdown.R [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
   check_is_package("use_pkgdown()")
-  edit_file(proj_path("_pkgdown.yml"))
-  use_build_ignore("_pkgdown.yml")
+  edit_file(proj_path(config_file))
+  use_build_ignore(config_file)
 
-  use_directory("docs", ignore = TRUE)
+  use_directory(dst_path, ignore = TRUE)
```
Comment (hadley) [inline R/pkgdown.R]: I think we should probably build ignore the docs directory, but not create it. That would also fix #495

File: R/pkgdown.R (hunk 1 above)
Comment (jayhesselberth) [reply R/pkgdown.R]: Yeah pkgdown takes:

```yaml
destination: docs
```

So we'd need to write that line into the created `_pkgdown.yml` file.

Comment (jayhesselberth) [reply R/pkgdown.R]: What's the recommended way to modify this file? Something like:

```r
cat(paste("destination:", dst_dir), file = config_file)
```

Comment (jennybc) [reply R/pkgdown.R]: Although it means something different, other functions in usethis this have a `destdir` argument FWIW (`create_*()`, `use_course()`).

Comment (hadley) [reply R/pkgdown.R]: @jayhesselberth  I think we can assume that this function creates the file, so you can use `write_over()`

And make the argument either `destination` (to match pkgdown) or `destdir` (to match usethis). (I don't have strong feelings which)

**Comment by hadley** [comment]
Can you please add a bullet to NEWS? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

(And looks like the docs need an update)

**Comment by hadley** [comment]
Thanks!

**Pull request merged**